dcos/subcommand.py

```diff
@@ -230,20 +303,31 @@ def _install_env(pkg, options):
     """
 
     pkg_dir = _package_dir(pkg.name())
+    resources = pkg.resource_json()
+    binary_cli = _get_cli_binary(resources)
 
-    install_operation = pkg.command_json()
-
-    env_dir = os.path.join(pkg_dir,
-                           constants.DCOS_SUBCOMMAND_VIRTUALENV_SUBDIR)
-
-    if 'pip' in install_operation:
-        _install_with_pip(
+    if binary_cli:
+        _install_with_binary(
             pkg.name(),
-            env_dir,
-            install_operation['pip'])
+            pkg_dir,
+            binary_cli)
     else:
-        raise DCOSException("Installation methods '{}' not supported".format(
-            install_operation.keys()))
+        install_operation = pkg.command_json()
```

Comment:
Yes, we should change the method signature to `has_cli_definition` and can check for `None` in `pkg.command_json` to see if that file exists. This won't catch the case where we only have the cli defined in `resource.cli` but not for the user's platform. In that case we will print `Installing cli subcommand...` and then `Could not find a CLI subcommand for your platform` which seems appropriate to me.